src/Microsoft.Diagnostics.Monitoring.WebApi/CollectionRuleStateHolder.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Globalization;
+
+namespace Microsoft.Diagnostics.Monitoring.WebApi
+{
+    internal class CollectionRuleStateHolder
```

Comment:
Conside record for automatic copying.

Reply:
I was having an issue locking that appeared after switching to using `record`, so I'm currently omitting it.